exporter/containerimage/writer.go

```diff
@@ -214,10 +237,51 @@ func (ic *ImageWriter) exportLayers(ctx context.Context, refCfg cacheconfig.RefC
 	return out, err
 }
 
+func (ic *ImageWriter) exportAttestations(ctx context.Context, s session.Group, desc *ocispecs.Descriptor, attestations ...exporter.Attestation) ([]InTotoAttestation, error) {
+	span, ctx := tracing.StartSpan(ctx, "export attestations")
+
+	eg, ctx := errgroup.WithContext(ctx)
+	attestationsDone := oneOffProgress(ctx, "exporting attestations")
+
+	out := make([]InTotoAttestation, len(attestations))
+
+	for i, attestation := range attestations {
+		mount, err := attestation.PredicateRef.Mount(ctx, true, s)
+		if err != nil {
+			return nil, err
+		}
+
+		lm := snapshot.LocalMounter(mount)
+		src, err := lm.Mount()
+		if err != nil {
+			return nil, err
+		}
+		defer lm.Unmount()
+
+		rawPredicate, err := os.ReadFile(path.Join(src, attestation.PredicatePath))
+		if err != nil {
+			return nil, err
+		}
+		out[i] = NewAttestation(attestation.PredicateType, rawPredicate)
+		for _, s := range attestation.Subjects {
+			switch s2 := s.(type) {
+			case *exporter.AttestationSubjectSelf:
+				out[i].AddSubject(platforms.Format(*desc.Platform)+" image", desc.Digest)
```

Comment:
I'm not sure if this name value in here makes sense? Any examples where it is done like this?

Reply:
I am also not sure about this name. I thought it needs to be a PURL like (`"pkg:docker/index.docker.io/library/ubuntu@sha256:f4f9ed916ce66b49307c42651ad6335096e02a78bb3585888103131f9cccd663?arch=amd64\u0026os=linux"`), or perhaps it can be unspecified as we cannot quite be sure of what all the tags that might be used in the future... I guess `"name": "_"` was what you thought of using for that purpose?

Reply:
> We do need to try and keep the name stable between buildkit runs though if possible - so we definitely don't want a digest of the image here.

So PURL example above is not quite suitable? Unfortunately that is what the spec seems to mandate at present...

Reply:
>  so we definitely don't want a digest of the image here.

What would be the problem with digest as digest should be stable? I do see a problem with name, though, as there can be multiple names and images get transferred etc. Maybe just `pkg:docker@digest` 🤷‍♂️ ?

Reply:
Ah I think I've misread "stable" as stable between builds, where it only says as stable between attestations.

In that case digests seem fine to me. If we allow splitting the attestation manifest later, we just need to make sure that it has the same subject name later.

Reply:
The spec doesn't mandate PURL. They've explicitly provided an example with subject names like `curl-7.72.0.tar.bz2` [here](https://github.com/in-toto/attestation/blob/main/README.md#provenance-example).

By my interpretation, the subject name needs to be suitably unique, but I think it should probably also convey some information that is human readable. There's not really much point in duplicating the digest IMO, since that's either part of the `digest` field (for the Self reference), or referencable through the proposal f descriptor annotation.

Reply:
Perhaps PURL is rather highly recommend within InToto ecosystem, I certainly read that somewhere.

Reply:
To summarize the options:
- `name` = `digest`. This duplicates information, and the name isn't particularly informative of what the content is.
- `name` = `tag`. This isn't always true, since images might move around.
- `name` = `arch`. Duplicates information.
- `name` = url based on the above. Same problems as all the above.
- `name` = `_`. No information contained here, we could maybe start with this and change later?

I'm going to go with `_` option for now, and we can follow up later if necessary.